source/tutorials/packaging-existing-software.md

```diff
@@ -372,30 +428,97 @@ error: builder for '/nix/store/bw2d4rp2k1l5rg49hds199ma2mz36x47-icat.drv' failed
        For full logs, run 'nix log /nix/store/bw2d4rp2k1l5rg49hds199ma2mz36x47-icat.drv'.
 ```
 
-You can see a few warnings which should be corrected in the upstream code, but the important bit for this tutorial is `fatal error: X11/Xlib.h: No such file or directory`: another dependency is missing.
+You can see a few warnings which should be corrected in the upstream code.
+But the important bit for this tutorial is `fatal error: X11/Xlib.h: No such file or directory`: another dependency is missing.
 
+## Finding packages
 
-:::{note}
-Determining from where to source a dependency is currently a somewhat-involved process: it helps to become familiar with searching the `nixpkgs` source for keywords.
+Determining from where to source a dependency is currently a somewhat involved, because package names don't always correspond to library or program names.
 
-Consider using `nix-locate` from the [`nix-index`](https://github.com/nix-community/nix-index) tool to find derivations that provide what you need.
+:::{tip}
+The easiest way to find what you need is on search.nixos.org/packages.
 :::
 
+### search.nixos.org
+
+Unfortunately in this case, [searching for `x11`](https://search.nixos.org/packages?query=x11) produces too many irrelevant results because X11 is ubiquitous.
+On the left side bar there is a list package sets, and [selecting `xorg`](https://search.nixos.org/packages?channel=23.11&buckets={%22package_attr_set%22%3A[%22xorg%22]%2C%22package_license_set%22%3A[]%2C%22package_maintainers_set%22%3A[]%2C%22package_platforms%22%3A[]}&query=x11) shows something promising.
+
+In case all else fails, it helps to become familiar with searching the [Nixpkgs source code](https://github.com/nixos/nixpkgs) for keywords.
+
+### Git and `rg`
+
+To find name assignments in the source, search for `"<keyword> ="`.
+For example, these are the search results for [`"x11 = "`](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22x11+%3D%22&type=code) or [`"libx11 ="`](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22libx11+%3D%22&type=code) on Github .
+
+Or fetch a local clone of the repository and use `rg`.
+Nixpkgs is huge.
+Only clone the latest revision if you don't want to wait a long time:
+
+```console
+$ nix-shell -p git ripgrep
+[nix-shell:~]$ git glone https://github.com/NixOS/nixpkgs --depth 1
+```
+Nixpkgs is *huge*, therefore specify which subdirectory you want to search:
```

Comment:
I don't think we need to emphasize its size twice within the same paragraph :smile:

Reply:
Yes, a failed attempt to be funny.